### PR TITLE
Add mod_caps_rdbms

### PR DESCRIPTION
**Pull request by arcusfelis**
This PR addresses MIM-2067.

Proposed changes include:
* module, tests
* using jiffy to encode, because features could be an integer according to the type spec...

We do not try to refactor/optimize or make reasonable caps logic in this PR though, just persist data in RDBMS instead of mnesia here. For example, data removal is not tested and it is not called from anywhere in the main module to begin with...

### CODE REVIEW

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/205369) / elasticsearch_and_cassandra_mnesia / 165afab80fd26aa1ba2d49997d0a0dcccb5e484c
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4211/205369/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4211/205369/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40b2d63128bd15.2024-01-14_19.36.58/index.html)
**OK: 382** / Failed: 0 / **User-skipped: 40** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/205444) / elasticsearch_and_cassandra_mnesia / 4e259b1b0d5f8b09c995a1c5137fef27ce0e03ea
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4211/205444/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4211/205444/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%4097de4a8186bf.2024-01-14_20.03.19/index.html)
**OK: 382** / Failed: 0 / **User-skipped: 40** / Auto-skipped: 0

**Review by NelsonVides — COMMENTED** [review]
It all seems fine, but if we're introducing schema changes, it'd be sure they're final, before then one day in the future we decide to optimise this and then we'll have to request our users to migrate data they already have. No need to ensure this now, but, would be good before a future release

File: big_tests/tests/pep_SUITE.erl [hunk 1]
```diff
@@ -504,14 +504,18 @@ field_spec({Var, Value}) when is_list(Value) -> #{var => Var, values => Value};
 field_spec({Var, Value}) -> #{var => Var, values => [Value]}.
 
 required_modules() ->
-    [{mod_caps, config_parser_helper:default_mod_config(mod_caps)},
+    HostType = domain_helper:host_type(),
+    Backend = mongoose_helper:get_backend_mnesia_rdbms(HostType),
+    [{mod_caps, config_parser_helper:mod_config(mod_caps, #{backend => Backend})},
```
[the remark this answers is not in the document]
Comment (arcusfelis) [reply big_tests/tests/pep_SUITE.erl:509]: sadly we do not, could have some `mod_config_with_auto_backend/2` somewhere...

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/205578) / elasticsearch_and_cassandra_mnesia / a6cd240d2215d2da68488513110c0f144e945453
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4211/205578/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4211/205578/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40cfb4fe069646.2024-01-15_14.28.13/index.html)
**OK: 382** / Failed: 0 / **User-skipped: 40** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_26](https://circleci.com/gh/esl/MongooseIM/205603) / elasticsearch_and_cassandra_mnesia / eba748be249b1093c6181f3592d4370b1c531c65
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4211/205603/elasticsearch_and_cassandra_mnesia.26.1.2-amd64)/ [big](https://circleci-mim-results.s3.eu-central-1.amazonaws.com/PR/4211/205603/elasticsearch_and_cassandra_mnesia.26.1.2-amd64/big/ct_run.test%40a8d9d4496874.2024-01-15_14.37.18/index.html)
**OK: 382** / Failed: 0 / **User-skipped: 40** / Auto-skipped: 0

**Comment by arcusfelis** [comment]
```
[2024-01-15T15:40:16.778Z] ['error'] There was an error running the uploader: Error uploading to https://codecov.io: Error: There was an error fetching the storage URL during POST: 503 - upstream connect error or disconnect/reset before headers. reset reason: connection failure
```
... codecov fails again

File: src/mod_caps_rdbms.erl [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+read(HostType, {Node, SubNode}) ->
+    case mongoose_rdbms:execute(HostType, caps_select, [Node, SubNode]) of
+        [{selected, [{Encoded}]}] -> {ok, jiffy:decode(Encoded)};
+        _ -> error
+    end.
+
+-spec write(mongooseim:host_type(), mod_caps:node_pair(),
+            mod_caps:maybe_pending_features()) -> ok.
+write(HostType, {Node, SubNode}, Features) ->
+    Encoded = jiffy:encode(Features),
```
Comment (NelsonVides) [inline src/mod_caps_rdbms.erl:34]: I'm amazed jiffy just works here 🙄

**Review by NelsonVides — APPROVED** [review]
LGTM (for now xD)

**Pull request merged**